### ISSUE TITLE
Plan an outline for this guide

### ISSUE THREAD

**Issue by seanmadsen** [civicrm/civicrm-sysadmin-guide#1]
I am proposing the following outline for this guide and I'd love to get feedback from others:

* Home
* Planning your site *(migrated from [this content](https://docs.civicrm.org/user/en/latest/getting-prepared/is-civicrm-for-you/) in the User Guide)*
    * Identifying your needs
    * Trying a demo
    * Hosting
    * Choosing a CMS
    * Transitioning to CiviCRM
* Requirements
* Installation
    * Install in Drupal 8
    * Install in Drupal 7
    * Install in Drupal 6
    * Install in WordPress
    * Install in Joomla
    * Install in Backdrop
    * Multi-site installation
* Setup
    * Setup *(including: checklist, etc.)*
    * Backups *(including: backup, restore)*
    * Monitoring *(including: System status, recommended sysadmin monitoring tools)*
    * Securing your site *(including: CMS Permissions, Civi ACLs, CAPTCHA, SSL, PHPIDS)*
    * Data migration
    * Sync users and contacts
    * Scheduled jobs *(including: scheduled jobs, Cron job setup)*
    * CiviMail setup *(including: content from [Andrew's blog post](https://civicrm.org/blog/andrewhunt/alternatives-to-mandrill)*
    * SMS setup
    * Payment processors
    * Detailed logging
    * Troubleshooting
* Upgrades
    * Upgrades overview
    * Upgrade in Drupal 8
    * Upgrade in Drupal 7
    * Upgrade in Drupal 6
    * Upgrade in WordPress
    * Upgrade in Joomla
    * Upgrade in Backdrop
    * Upgrade troubleshooting
* Customization
    * Basic customization  *(mostly points to User Guide)*
    * Extensions *(including: installing, removing, upgrading extensions. Also will point to the Dev Guide)*
    * Other languages
    * Drupal integration
        * *(sub-pages will eventually here, but will not be covered by MIH campaign)*
    * Backdrop integration
    * Wordpress integration
    * Joomla integration
    * Custom development
* Special situations
    * Changing installation directories
    * Changing domain names
    * Switching servers
    * Switching CMS platforms
    * Data redaction

**Comment by nganivet** [comment]
Whoa, that is very complete and it seems like a huge endeavor to write 
all of this (could be a published book in and of itself!). Would it be 
possible to identify priority content and fill that in first, and then 
move to lower priority items. In particular the section about backups / 
monitoring / security are kinda non-CiviCRM specific and will be quite 
voluminous if you want to get into any specifics.

On the fence about having paragraphs for CMS's intertwined with content, 
or having separate guides/sections for each CMS'es with the same 
headings. The later will allow for an easier read since a WordPress 
person can the just read the common generic paragraphs in the overall 
guide, and then dive into the WordPress-specific guide. Otherwise he/she 
will always jump from section to section in the larger guide. This would 
also help reduce the length of these guides to something more 
manageable.

------ Original Message ------

**Comment by seanmadsen** [comment]
Thanks @nganivet 

Most of this content already exists. It's true that some of it (e.g. monitoring) doesn't exist yet, but I think it's useful to visualize the structure of a complete guide. To me, the highest priority content is the content in the wiki, so I want to begin there.

I chose to organize by topic primarily and CMS secondarily. It sounds like you're suggesting organizing by CMS primarily and topic secondarily. I'm open to that. I'd like to hear other's thoughts on it.

**Comment by JoeMurray** [comment]
I'd suggest lots of links for the Planning your site chapter - our Using CiviCRM has many dozens of pages.

I think Special Situations > Switching Servers is relatively common, and could be broadened to include changing install directory on same server and also changing the domain but not the server or install dir.

@nganivet I don't see this as a guide that will be read straight through by someone doing a particular CMS install, upgrade, customizations, etc. Were that so, it would be better to have top level headings by CMS. 

Should customizations go before Upgrades?

Other languages is not a customization, but a part of installing and upgrading.

What about Backdrop Integration?

**Comment by mlutfy** [comment]
> "Other languages is not a customization, but a part of installing and upgrading."

+1, don't forget that the translation files must be correctly copied to the server before starting the initial installation, as well as when upgrading (it's easy to forget the `l10n` directory when upgrading the code base).

However, there are more advanced topics for localization: mainly how to use 'native gettext' (I guess this could go either in "Customization" or "Setup", but that step can be done after installation).

**Comment by michaelmcandrew** [comment]
Looks great and well thought out...

**Comment by seanmadsen** [comment]
@JoeMurray 

> I'd suggest lots of links for the Planning your site chapter

For that chapter, I was planning to bring [this content](https://docs.civicrm.org/user/en/latest/getting-prepared/is-civicrm-for-you/) over from the User Guide.

--- 

> I think Special Situations ... could be broadened

Good suggestions! I modified the outline to include your suggested topics.

---

> Should customizations go before Upgrades?

I'm open to that. It's very easy to change that sort of thing later (since re-ordering things doesn't break any links). So for now, I'm going to leave this as-is and see if anyone else suggests it. 

My rationale for putting upgrades before customization is that everyone needs to perform upgrades, but not everyone needs to perform customization. And in some cases people will need to perform an upgrade before they ever customize anything. 

---

> Backdrop Integration?

Added.

---

> Other languages is not a customization, but a part of installing and upgrading.

Thanks for clarifying. And @mlutfy too. I've only ever used CiviCRM in English so this stuff is new to me. 

Can one of you suggest more specific changes to the outline that will better reflect these concepts?

**Comment by seanmadsen** [comment]
In the spirit of making progress through making decisions, I'm closing this ticket now and will be moving forward with the outline listed at the beginning of this ticket. 

If anyone has further comments, please feel free to make them here, even though this ticket is closed. I'm still open to tweaking the outline.
